os/arch/arm/src/common/up_usestack.c

```diff
@@ -156,7 +133,11 @@ int up_use_stack(struct tcb_s *tcb, void *stack, size_t stack_size)
 	 * referenced as positive word offsets from sp.
 	 */
 
+#ifdef CONFIG_ARCH_ARMV7A_FAMILY
+	top_of_stack = (uint32_t)tcb->stack_alloc_ptr + stack_size;
+#else
 	top_of_stack = (uint32_t)tcb->stack_alloc_ptr + stack_size - 4;
+#endif
```

Comment:
This is a common file. It looks not good to add archi specific code here.
How about defining new config for this?
1. new config to on / off for -4
```
in up_userstack.c
#ifdef CONFIG_NEW
   top_.._stack = ....;
#else
   top_.._stack = .... - 4;
#endif

in Kconfig
config ARMV7A
    select NEW
```
2. new config to give adj value
```
in c
    top_...stack = .... + stack_size - CONFIG_NEW;

in defconfig
CONFIG_NEW=4
```

Reply:
And I can't find why -4 is unnecessary. Could you leave comment?

Reply:
hi @sunghan-chang 
I have been looking for a reason for this. But, as of now I have not been able to find out the exact reason. 
However, it looks like all cortex m code requires the -4 and cortex a32 does not need the -4.
If I try to change this in anyway, it is resulting in various crashes. 
So, I would like to keep this code for now and we will look into the reasoning for this at a later time. I think this can be a low priority activity.

Reply:
> This is a common file. It looks not good to add archi specific code here. How about defining new config for this?
> 
> 1. new config to on / off for -4
> 
> ```
> in up_userstack.c
> #ifdef CONFIG_NEW
>    top_.._stack = ....;
> #else
>    top_.._stack = .... - 4;
> #endif
> 
> in Kconfig
> config ARMV7A
>     select NEW
> ```
> 
> 2. new config to give adj value
> 
> ```
> in c
>     top_...stack = .... + stack_size - CONFIG_NEW;
> 
> in defconfig
> CONFIG_NEW=4
> ```

I am not sure what to name the CONFIG_NEW because I am not able to find out the reason for this.